src/xenia/config.cc

```diff
@@ -0,0 +1,101 @@
+#include "xenia/config.h"
+#include "base/filesystem.h"
+#include "third_party/cpptoml/include/cpptoml.h"
+#include "xenia/base/string.h"
+
+namespace xe {
+
+Config* Config::instance_;
+Config::Config(const std::wstring& file_path) : file_path_(file_path) {
+  if (!xe::filesystem::PathExists(file_path)) {
+    xe::filesystem::CreateFile(file_path);
+  }
+  LoadDefaults();
+  LoadConfig();
+  SaveConfig();
+}
+
+Config::~Config() { SaveConfig(); }
+
+void Config::Init(const std::wstring& content_path) {
+  auto file_path = xe::join_paths(content_path, L"config.toml");
+  instance_ = new Config(file_path);
+}
+void Config::Shutdown() {
+  delete instance_;
+  instance_ = nullptr;
+}
+
+void Config::LoadDefaults() {
+  gpu_ = "any";
```

Comment:
This won't be sustainable with the amount of options Xenia will eventually have.
Plan to create a static registration system in the future, ala `ConVar` in Source Engine.